### ISSUE TITLE
[Bug?]: Segmentation fault on x86_64 linux on yarn install with minimal package.json (yarn@3.5.1)

### ISSUE THREAD

**Issue by tisba** [yarnpkg/berry#5420]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

Initially encountered as part of a bigger project, but I was able to boil it down to a minimal example (see below).

`yarn install` started to segfault at some time using Docker. I haven't been able to pin point exactly when this started to happen, as the usage of Node/yarn is rather minimal.

On macOS 13.3.1 (22E261) with Node v20.0.0 installation succeeds.

### To reproduce

Using a minimal `package.json` and the following script run via Docker, I can reliably reproduce the crash.

I used `@rails/ujs` as this was the first dependency in the bigger project, that has no transient dependencies. As far as I can tell, it doesn't matter what dependency is specified, as long as there is at least one.

```bash
#!/bin/sh

# run via
#
#   docker run -it --rm -v "$(pwd)/test.sh":/app/test.sh:ro -w /app node:18.16 /app/test.sh

set -e

cat << EOF > ./package.json
{
    "packageManager": "yarn@3.5.1",
    "dependencies": {
        "@rails/ujs": "7.0.4"
    }
}
EOF

corepack enable

npm install -g npm@9.6.5

printf "uname -a: %s\n" "$(uname -a)"
printf "Node version: %s\n" "$(node --version)"
printf "Yarn version: %s\n" "$(yarn --version)"
printf "npm version: %s\n" "$(npm --version)"
printf "\n\n"

printf "Running: yarn install\n"

yarn install
```

### Environment

`yarn dlx -q envinfo --preset jest` crashes as well. Here is the output from my script above regarding the env:

```shell
uname -a: Linux 275135397ae6 5.15.49-linuxkit #1 SMP Tue Sep 13 07:51:46 UTC 2022 x86_64 GNU/Linux
Node version: v18.16.0
Yarn version: 3.5.1
npm version: 9.6.5
```

Since I'm running this in docker:

```shell
$ docker version
Client:
 Cloud integration: v1.0.31
 Version:           23.0.5
 API version:       1.42
 Go version:        go1.19.8
 Git commit:        bc4487a
 Built:             Wed Apr 26 16:12:52 2023
 OS/Arch:           darwin/amd64
 Context:           default

Server: Docker Desktop 4.19.0 (106363)
 Engine:
  Version:          23.0.5
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.19.8
  Git commit:       94d3ad6
  Built:            Wed Apr 26 16:17:45 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.20
  GitCommit:        2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc:
  Version:          1.1.5
  GitCommit:        v1.1.5-0-gf19387a
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Additional context

I also get this failure with a `node:20` docker image. I also tried various versions of npm and yarn.

<img width="849" alt="CleanShot 2023-05-02 at 17 13 19@2x" src="https://user-images.githubusercontent.com/75156/235708982-5253e89b-9bb2-479d-b37e-d5b71b8de672.png">

**Comment by tisba** [comment]
Looks like this might be a macOS x86 Docker Desktop specific problem, see https://github.com/docker/for-mac/issues/6824.

**Comment by tisba** [comment]
I cannot verify (since I don't have any x86-based macOS anymore), but apparently this has been fixed in Docker 4.30.0 (https://docs.docker.com/desktop/release-notes/#4300). See also https://github.com/docker/for-mac/issues/6824#issuecomment-2096417353.
